keeper-gradle-plugin/src/test/kotlin/com/slack/keeper/KeeperFunctionalTest.kt

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (C) 2020 Slack Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.slack.keeper
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.javapoet.ClassName
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.intellij.lang.annotations.Language
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+import java.io.File
+import java.util.zip.ZipFile
+import javax.lang.model.element.Modifier.STATIC
+import com.squareup.kotlinpoet.ClassName as KpClassName
+
+/**
+ * Testing gradle plugins is finicky. If you get errors when running from the IDE, try following
```

Comment:
Never had a problem, myself!

Reply:
That's obviously because there's a secret `if (user.isCurrentOrFormerGradleEmployee) skipFailures())` method

Reply:
There's also the `—Dorg.gradle.suckit` system property.